### ISSUE TITLE
Issue in finding the least-cost paths to create the MPG : Some LCP are actually not the LCP ?

### ISSUE THREAD

**Issue by Klemet** [achubaty/grainscape#72]
Hello there 😄 ! Sorry to come yet again with an issue.

To make it simple : I found a case where Grainscape seemed to have created a link between two patches via the `MPG` function that does not actually correspond to the least-cost path between the two.

See [GrainscapeIssue.zip](https://github.com/user-attachments/files/16459877/GrainscapeIssue.zip) for sample files.

Here is the resistance map (yellow = cost of 1 for forests, green = cost of 1000 for lakes) + the patches (white) : 
![image](https://github.com/user-attachments/assets/6f2dfc84-abc6-45ae-9ac8-dd46e6dba16b)

Here is the resulting MPG, and a highlight of the two patches with the problematic connection : 
![image](https://github.com/user-attachments/assets/471d29f7-4726-41db-91e9-f8f8dd89c23e)

Zooming in in QGIS, we can see that the obvious least-cost path is a path with a final cost of `3` through three cells with a cost of `1` (displayed in purple on this screenshot, wtith the patches in green and the lakes with cost 1000 in yellow) :
![image](https://github.com/user-attachments/assets/7c41d6e5-c25b-434e-83f4-de067c491529)

However, the least-cost path indicated by the `MPG` object (see `lcpPerimWeight` raster) goes straight through the lake, giving a path of 1 cell, but with cost `1000` (in orange on this screenshot) :
![image](https://github.com/user-attachments/assets/c85863ce-3933-4bec-b745-d54e5f54c463)

I don't think this questions the structure of the MPG, but only the weight of some particular links. I noticed it while doing an analysis similar to what Linkage Mapper does with corridors, where the pixels of the corridor maps correspond to "the additional cost of going through this pixel compared to using the least-cost path between the two patches linked by the corridor". In that case, because some least-cost path were wrong (I had several cases in my landscape), this resulted in negative values for this corridors (which should not happen).

No idea as to why this happens, and I haven't identified a pattern in this errors that could give me a pointer where the LCP algorithm of Grainscape might have an issue. I can post other cases if needed.

**Comment by achubaty** [comment]
Thanks for reporting this. I will try to get to this as soon as possible, but unlikely before September. (I'm hoping to get updated version of this package to CRAN this fall.)

@ecologics do you have time to investigate before then?

**Comment by Klemet** [comment]
There's no rush on my side; thank you for taking a look at it !

**Comment by achubaty** [comment]
@Klemet I can reproduce this, even zoomed into (i.e., cropped to) the problem area highlighted. I'm taking a look at this now, and hope to have it fixed in a few days.

**Comment by Klemet** [comment]
Thank you for the update, @achubaty !! It's very interesting to see how this came to be. 

Also, just wanted to say thanks for your work on Grainscape, and for you to not feel in a rush. I'm already very grateful of everything I was able to do with the package, and for the attention you're giving to this fix 🙏

Good luck for finding a fix !

**Comment by achubaty** [comment]
@Klemet apologies for the long wait on this! The fix (and a ton of other changes) are now in the development branch, to be merged into main and submitted to CRAN in the next few weeks. I'm holding off submitting right away because there are several breaking changes in this upcoming v1.0.0 release (see #77).

**Comment by Klemet** [comment]
Thanks Alex, that's wonderful ! Congrats for the upcoming update 😁 !
